designs/dotnetcore-debugging.md

```diff
@@ -0,0 +1,233 @@
+# .NET Core 2.0 / 2.1 Debugging
+
+This is a design for the .NET Core 2.0 and 2.1 missing debugging feature.
+
+### What is the problem
+
+Currently SAM CLI does not provide debugging support for .NET Core 2.0 and 2.1 because `dotnet` command does **not** support _start and wait for the debugger_ configuration out of the box and neither does current AWS Lambda runner ([2.0](https://github.com/lambci/docker-lambda/tree/master/dotnetcore2.0) / [2.1](https://github.com/lambci/docker-lambda/tree/master/dotnetcore2.1)) has debugging support flag or anything.
```

Comment:
There is some talk here that it might make it to 3.0

https://github.com/dotnet/core-setup/issues/198

Reply:
Yes, I've mentioned this issue in the document too. But I don't think it has some priority there now